### ISSUE TITLE
[iOS] App Stuck at Boot on Real Devices

### ISSUE THREAD

**Issue by IAmMarcellus** [facebook/react-native#26336]
TL;DR: My app gets stuck at the splash screen with no errors when I run on a real device ever since I upgraded to react native 60.4 from 59.5.  I even created a new project and copied the necessary files over, but it's still happening. However, it does run on the simulator.

## The Problem
When I run my app on a real iOS device from xcode, the app gets stuck on the splash screen and never loads, nor does it crash. If I close the app and then try to open it again, I get a blank white screen (and I've noticed that the packager doesn't even try to load the bundle again, which it should). This happens no matter how many times I try to reopen the app.
If I run the app on a real iOS device, but instead launch it from the command line using "run-ios --device", I can usually get it to work, but I may have to close the app and open it a few times.
Running it on the simulator works as expected for either launch method.
 
2 things that might be related:
1. running `react-native log-ios` doesn't work on real devices either. However, it does work on the simulator.
2. When running on a real device, the green loading bar at the top of the screen during boot isn't there, which it normally would be.

## Info
React Native version: 0.60.4
React Native Info:

System:
    OS: macOS 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
    Memory: 615.95 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.8.0 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.10.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
    Android SDK:
      API Levels: 27, 28, 29
      Build Tools: 27.0.3, 28.0.3, 29.0.0
      System Images: android-27 | Intel x86 Atom, android-27 | Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Intel x86 Atom, android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom_64, android-29 | Intel x86 Atom, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom_64, android-29 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom_64
      Android NDK: 20.0.5594570
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5522156
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6 
    react-native: 0.60.5 => 0.60.5 
  npmGlobalPackages:
    react-native-cli: 2.0.1

## What I've tried
I thought that it might be related to https://github.com/facebook/react-native/issues/24023 because they were experience differences between launching from Xcode vs `run-ios` and also dealing with an app stuck at boot, however, my app wasn't crashing after hanging like his was, and I've tried all of the suggestions there and none of them worked.
I've also tried adding additional App Transport Security allowances to the info.plist, but none of those seemed to help either.
And I've tried cleaning every single cache that might possibly be storing any react-native data.
## Steps To Reproduce

Unfortunately, I wasn't unable to reproduce this in a new RN project, which, because I'm using a newly initialized project that I copied old code to, makes me think it might be a cache problem (the current app even uses the same Bundle ID as the old one). However, I'm pretty sure I've cleared every cache possible and it hasn't changed the results. Likewise, the app is working fine on the simulator and was working fine before I updated to 0.60, so I hesitate to think its the old code, but rather is an environment issue, but I can't seem to narrow it down any further than that.

I've been beating my head over this issue for over a week and now it's delaying a crucial update for my app, please help!

**Comment by IAmMarcellus** [comment]
The following is what's logged in the Xcode console when I try and launch through it (I deleted a few warnings about require cycles that have always been showing up):

2019-09-04 22:25:25.306757-0400 Pulse[2144:1421808] [DYMTLInitPlatform] platform initialization successful
2019-09-04 22:25:25.475400-0400 Pulse[2144:1421804]  - <AppMeasurement>[I-ACS036002] Analytics screen reporting is enabled. Call +[FIRAnalytics setScreenName:setScreenClass:] to set the screen name or override the default screen class name. To disable screen reporting, set the flag FirebaseScreenReportingEnabled to NO (boolean) in the Info.plist
2019-09-04 22:25:25.545367-0400 Pulse[2144:1421653] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2019-09-04 22:25:25.546285-0400 Pulse[2144:1421653] [MC] Reading from public effective user settings.
2019-09-04 22:25:25.612043-0400 Pulse[2144:1421653] [Crashlytics] Version 3.13.4 (143)
2019-09-04 22:25:25.646506-0400 Pulse[2144:1421831] [NetworkInfo] Signal strength query returned error: Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied", descriptor: <CTServiceDescriptor 0x2831a6740, domain=1, instance=1>
2019-09-04 22:25:25.676534-0400 Pulse[2144:1421653] FBSDKLog: Unable to find a valid UIWindow
2019-09-04 22:25:25.701365-0400 Pulse[2144:1421798] 6.7.0 - [Firebase/Performance][I-PRF700001] Firebase Performance Diagnostics have been enabled!
2019-09-04 22:25:25.702329-0400 Pulse[2144:1421798] 6.7.0 - [Firebase/Analytics][I-ACS023007] Analytics v.60101000 started
2019-09-04 22:25:25.703878-0400 Pulse[2144:1421798] 6.7.0 - [Firebase/Analytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
2019-09-04 22:25:25.761 [info][tid:main][RCTCxxBridge.mm:236] Initializing <RCTCxxBridge: 0x105f31080> (parent: <RCTBridge: 0x281a86e60>, executor: (null))
2019-09-04 22:25:25.762217-0400 Pulse[2144:1421653] Initializing <RCTCxxBridge: 0x105f31080> (parent: <RCTBridge: 0x281a86e60>, executor: (null))
2019-09-04 22:25:25.846 [info][tid:main][RCTRootView.m:293] Running application Pulse ({
    initialProps =     {
    };
    rootTag = 1;
})
2019-09-04 22:25:25.846492-0400 Pulse[2144:1421653] Running application Pulse ({
    initialProps =     {
    };
    rootTag = 1;
})
2019-09-04 22:25:25.879445-0400 Pulse[2144:1421823] [NetworkInfo] Signal strength query returned error: Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied", descriptor: <CTServiceDescriptor 0x2831c4b00, domain=1, instance=1>
2019-09-04 22:25:25.935492-0400 Pulse[2144:1421834] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C2.1:2][0x106805e70] get output frames failed, state 8196
2019-09-04 22:25:25.935659-0400 Pulse[2144:1421834] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C2.1:2][0x106805e70] get output frames failed, state 8196
2019-09-04 22:25:25.936065-0400 Pulse[2144:1421834] TIC Read Status [2:0x0]: 1:57
2019-09-04 22:25:25.936086-0400 Pulse[2144:1421834] TIC Read Status [2:0x0]: 1:57
2019-09-04 22:25:25.936517-0400 Pulse[2144:1421834] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C1.1:2][0x105f125a0] get output frames failed, state 8196
2019-09-04 22:25:25.936662-0400 Pulse[2144:1421834] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C1.1:2][0x105f125a0] get output frames failed, state 8196
2019-09-04 22:25:25.937110-0400 Pulse[2144:1421834] TIC Read Status [1:0x0]: 1:57
2019-09-04 22:25:25.937276-0400 Pulse[2144:1421834] TIC Read Status [1:0x0]: 1:57
2019-09-04 22:25:25.938357-0400 Pulse[2144:1421798] [Fabric] failed to download settings Error Domain=FABNetworkError Code=-5 "(null)" UserInfo={status_code=403, type=2, request_id=8afe684442168deb81db937f4ded0667, content_type=application/json; charset=utf-8}
2019-09-04 22:25:25.989040-0400 Pulse[2144:1421831] 6.7.0 - [Firebase/Analytics][I-ACS025026] Deferring to Firebase for event data collection
2019-09-04 22:25:26.065879-0400 Pulse[2144:1421653] -[RNFirebase init] [Line 18] Setting up RNFirebase instance
2019-09-04 22:25:26.081990-0400 Pulse[2144:1421841] Metal GPU Frame Capture Enabled
2019-09-04 22:25:26.086563-0400 Pulse[2144:1421841] Metal API Validation Enabled
2019-09-04 22:25:26.441727-0400 Pulse[2144:1421824] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C6.1:2][0x105f19050] get output frames failed, state 8196
2019-09-04 22:25:26.470521-0400 Pulse[2144:1421824] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C6.1:2][0x105f19050] get output frames failed, state 8196
2019-09-04 22:25:26.470898-0400 Pulse[2144:1421824] TIC Read Status [6:0x0]: 1:57
2019-09-04 22:25:26.471591-0400 Pulse[2144:1421824] TIC Read Status [6:0x0]: 1:57
2019-09-04 22:25:26.710976-0400 Pulse[2144:1421850] WF: === Starting WebFilter logging for process Pulse
2019-09-04 22:25:26.711029-0400 Pulse[2144:1421850] WF: _userSettingsForUser mobile: {
    filterBlacklist =     (
    );
    filterWhitelist =     (
    );
    noOverridingAllowed = 1;
    restrictWeb = 1;
    useContentFilter = 0;
    useContentFilterOverrides = 0;
    whitelistEnabled = 0;
}
2019-09-04 22:25:26.711073-0400 Pulse[2144:1421850] WF: _WebFilterIsActive returning: NO
2019-09-04 22:25:40.394871-0400 Pulse[2144:1421653] [NetworkInfo] Signal strength query returned error: Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied", descriptor: <CTServiceDescriptor 0x2831f0040, domain=1, instance=1>
2019-09-04 22:25:40.611568-0400 Pulse[2144:1421804] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C13.1:2][0x105f6f490] get output frames failed, state 8196
2019-09-04 22:25:40.611963-0400 Pulse[2144:1421804] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C13.1:2][0x105f6f490] get output frames failed, state 8196
2019-09-04 22:25:40.612679-0400 Pulse[2144:1421804] TIC Read Status [13:0x0]: 1:57
2019-09-04 22:25:40.612821-0400 Pulse[2144:1421804] TIC Read Status [13:0x0]: 1:57
2019-09-04 22:25:40.794012-0400 Pulse[2144:1421831] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C14.1:2][0x105f738d0] get output frames failed, state 8196
2019-09-04 22:25:40.794332-0400 Pulse[2144:1421831] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C14.1:2][0x105f738d0] get output frames failed, state 8196
2019-09-04 22:25:40.794812-0400 Pulse[2144:1421831] TIC Read Status [14:0x0]: 1:57
2019-09-04 22:25:40.794921-0400 Pulse[2144:1421831] TIC Read Status [14:0x0]: 1:57
2019-09-04 22:25:40.835118-0400 Pulse[2144:1421804] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C12.1:2][0x106871690] get output frames failed, state 8196
2019-09-04 22:25:40.835475-0400 Pulse[2144:1421804] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C12.1:2][0x106871690] get output frames failed, state 8196
2019-09-04 22:25:40.835966-0400 Pulse[2144:1421804] TIC Read Status [12:0x0]: 1:57
2019-09-04 22:25:40.836032-0400 Pulse[2144:1421804] TIC Read Status [12:0x0]: 1:57
2019-09-04 22:26:27.214581-0400 Pulse[2144:1421831] [] nw_socket_handle_socket_event [C16:1] Socket SO_ERROR [61: Connection refused]
2019-09-04 22:26:27.214978-0400 Pulse[2144:1421843] [] nw_connection_get_connected_socket [C16] Client called nw_connection_get_connected_socket on unconnected nw_connection
2019-09-04 22:26:27.214996-0400 Pulse[2144:1421843] TCP Conn 0x28049c480 Failed : error 0:61 [61]
2019-09-04 22:26:27.515 [warn][tid:com.facebook.react.JavaScript] Accessing view manager configs directly off UIManager via UIManager['getConstants'] is no longer supported. Use UIManager.getViewManagerConfig('getConstants') instead.
2019-09-04 22:26:27.515505-0400 Pulse[2144:1421845] Accessing view manager configs directly off UIManager via UIManager['getConstants'] is no longer supported. Use UIManager.getViewManagerConfig('getConstants') instead.
2019-09-04 22:26:27.556 [info][tid:com.facebook.react.JavaScript] Running application "Pulse" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
2019-09-04 22:26:27.556602-0400 Pulse[2144:1421845] Running application "Pulse" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
2019-09-04 22:26:27.889 [warn][tid:com.facebook.react.JavaScript] Warning: componentWillMount is deprecated and will be removed in the next major version. Use componentDidMount instead. As a temporary workaround, you can rename to UNSAFE_componentWillMount.

Please update the following components: AnimatedComponent

Learn more about this warning here:
https://fb.me/react-async-component-lifecycle-hooks
2019-09-04 22:26:27.890867-0400 Pulse[2144:1421845] Warning: componentWillMount is deprecated and will be removed in the next major version. Use componentDidMount instead. As a temporary workaround, you can rename to UNSAFE_componentWillMount.

Please update the following components: AnimatedComponent

Learn more about this warning here:
https://fb.me/react-async-component-lifecycle-hooks
2019-09-04 22:26:28.085 [warn][tid:com.facebook.react.JavaScript] Warning: componentWillReceiveProps is deprecated and will be removed in the next major version. Use static getDerivedStateFromProps instead.

Please update the following components: withAnimatable(View)

Learn more about this warning here:
https://fb.me/react-async-component-lifecycle-hooks
2019-09-04 22:26:28.085706-0400 Pulse[2144:1421845] Warning: componentWillReceiveProps is deprecated and will be removed in the next major version. Use static getDerivedStateFromProps instead.

Please update the following components: withAnimatable(View)

Learn more about this warning here:
https://fb.me/react-async-component-lifecycle-hooks
2019-09-04 22:26:28.471450-0400 Pulse[2144:1421653] <Google> To get test ads on this device, set: request.testDevices = @[ @"4326293742158daec0237f3764fdb77d" ];
2019-09-04 22:26:28.522 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x105f31ca0; reactTag: 1; frame = (0 0; 375 667); gestureRecognizers = <NSArray: 0x283fd1800>; layer = <CALayer: 0x2831ce5e0>>
2019-09-04 22:26:28.522001-0400 Pulse[2144:1421653] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x105f31ca0; reactTag: 1; frame = (0 0; 375 667); gestureRecognizers = <NSArray: 0x283fd1800>; layer = <CALayer: 0x2831ce5e0>>
2019-09-04 22:27:26.381613-0400 Pulse[2144:1422092] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C5.1:2][0x105f2d270] get output frames failed, state 8196
2019-09-04 22:27:26.382599-0400 Pulse[2144:1422092] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C5.1:2][0x105f2d270] get output frames failed, state 8196
2019-09-04 22:27:26.384039-0400 Pulse[2144:1422092] TIC Read Status [5:0x0]: 1:57
2019-09-04 22:27:26.384327-0400 Pulse[2144:1422092] TIC Read Status [5:0x0]: 1:57
2019-09-04 22:27:28.567934-0400 Pulse[2144:1422092] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C11.1:2][0x105f62ba0] get output frames failed, state 8196
2019-09-04 22:27:28.568935-0400 Pulse[2144:1422092] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C11.1:2][0x105f62ba0] get output frames failed, state 8196
2019-09-04 22:27:28.570692-0400 Pulse[2144:1422092] TIC Read Status [11:0x0]: 1:57
2019-09-04 22:27:28.570864-0400 Pulse[2144:1422092] TIC Read Status [11:0x0]: 1:57

**Comment by IAmMarcellus** [comment]
Update: I realized what makes the app hang, vs what doesn't. On a real device, that doesn't have the app installed already, running the app from Xcode will make it hang every time, but I press the stop button in Xcode, the app will exit and if I open it again, it seems to work fine every time. Then, once the app is installed (even if I didn't make it run the first time I installed it), I can clear all the caches and delete all the builds and everything and it will work fine. The only thing strange is that I never see the green loading thing while booting, ever.
This also means that the problem might be reproducible because when running a newly initiated problem to test my problem, the app closes after building and xcode stops running because I have to verify that its a trusted app.
So, I definitely think there is a bug, either in my environment or in react native, but is this a bug with the packager that's just causing it to hang on the first boot when launching from my local machine or would this continue to be a problem after release? Or rather, if I can archive this build without any error, is it safe to submit this build with these errors?

**Comment by IAmMarcellus** [comment]
I had shared the console messages that show when the app crashes in debug mode, but I don't think there was anything valuable in them so I hid the comment. However, these are the messages that show up in the console during release mode and I think they might actually be more informative:

019-09-05 01:42:48.141719-0400 Pulse[2230:1443246] void SendDelegateMessage(NSInvocation *): delegate (webView:connectionPropertiesForResource:dataSource:) failed to return after waiting 10.000000 seconds. main run loop mode: kCFRunLoopDefaultMode
2019-09-05 01:42:51.076041-0400 Pulse[2230:1443191] [BoringSSL] nw_protocol_boringssl_input_finished(1543) [C12.1:2][0x10f110470] Peer disconnected during the middle of a handshake. Sending errSSLClosedNoNotify(-9816) alert
2019-09-05 01:42:51.090792-0400 Pulse[2230:1443191] TIC TCP Conn Failed [12:0x280e79f80]: 3:-9816 Err(-9816)
2019-09-05 01:42:51.092419-0400 Pulse[2230:1443191] NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, -9816)
2019-09-05 01:42:51.092501-0400 Pulse[2230:1443191] Task <55AF8C23-3355-451C-913F-3BF881EC0E0A>.<1> HTTP load failed (error code: -1200 [3:-9816])
2019-09-05 01:42:51.093397-0400 Pulse[2230:1443191] Task <55AF8C23-3355-451C-913F-3BF881EC0E0A>.<1> finished with error - code: -1200
2019-09-05 01:42:58.143142-0400 Pulse[2230:1443246] void SendDelegateMessage(NSInvocation *): delegate (webView:connectionPropertiesForResource:dataSource:) failed to return after waiting 10.000000 seconds. main run loop mode: kCFRunLoopDefaultMode
2019-09-05 01:43:08.144568-0400 Pulse[2230:1443246] void SendDelegateMessage(NSInvocation *): delegate (webView:connectionPropertiesForResource:dataSource:) failed to return after waiting 10.000000 seconds. main run loop mode: kCFRunLoopDefaultMode
2019-09-05 01:43:18.145968-0400 Pulse[2230:1443246] void SendDelegateMessage(NSInvocation *): delegate (webView:connectionPropertiesForResource:dataSource:) failed to return after waiting 10.000000 seconds. main run loop mode: kCFRunLoopDefaultMode
2019-09-05 01:43:28.147356-0400 Pulse[2230:1443246] void SendDelegateMessage(NSInvocation *): delegate (webView:connectionPropertiesForResource:dataSource:) failed to return after waiting 10.000000 seconds. main run loop mode: kCFRunLoopDefaultMode
2019-09-05 01:43:38.148897-0400 Pulse[2230:1443246] void SendDelegateMessage(NSInvocation *): delegate (webView:connectionPropertiesForResource:dataSource:) failed to return after waiting 10.000000 seconds. main run loop mode: kCFRunLoopDefaultMode
2019-09-05 01:43:48.150332-0400 Pulse[2230:1443246] void SendDelegateMessage(NSInvocation *): delegate (webView:connectionPropertiesForResource:dataSource:) failed to return after waiting 10.000000 seconds. main run loop mode: kCFRunLoopDefaultMode

**Comment by vantb9x** [comment]
> This issue has to do with using modals and alerts at the same time on iOS. It is definitely a bug and something similar has been mentioned here: #10471. Since this issue doesn't relate to the actual problem (and since I solved my issue by changing the logic of my code), I'm going to close this. However, the bug definitely still exists.

Thank you so much. It works for me
